### PR TITLE
Do not assume a specific order of headers

### PR DESCRIPTION
**Pull request by simhnna**
The spec doesn't appear to define an order. I receive the headers in different order.

I needed to comment the code to work out how it works. After I've done that I refactored it a little so it hopefully is a little easier to read

### CODE REVIEW

**Comment by simhnna** [comment]
Here's the part of the RFC that says the order isn't relevant https://datatracker.ietf.org/doc/html/rfc2045#section-3

File: src/multipart.ts [hunk 1]
```diff
@@ -128,7 +148,7 @@ export function DemoData(): { body: Buffer; boundary: string } {
   body += '------WebKitFormBoundaryvef1fLxmoUdYZWXp--\r\n'
   return {
     body: Buffer.from(body),
-    boundary: '----WebKitFormBoundaryvef1fLxmoUdYZWXp'
+    boundary: '----WebKitFormBoundaryvef1fLxmoUdYZWXp',
```
Comment (nachomazzara) [inline src/multipart.ts:151]: Can we remove the trailing comma?

File: src/multipart.ts [hunk 2]
```diff
@@ -149,37 +169,37 @@ function process(part: Part): Input {
...
   if (filenameData) {
     input = obj(filenameData)
-    const contentType = part.info.split(':')[1].trim()
+    const contentType = part.contentTypeHeader.split(':')[1].trim()
     Object.defineProperty(input, 'type', {
       value: contentType,
       writable: true,
       enumerable: true,
-      configurable: true
+      configurable: true,
```
Comment (nachomazzara) [inline src/multipart.ts:187]: can we remove the trailing coma?

File: src/multipart.ts [hunk 3]
```diff
@@ -149,37 +169,37 @@ function process(part: Part): Input {
       value: b,
       writable: true,
       enumerable: true,
-      configurable: true
+      configurable: true,
```
Comment (nachomazzara) [inline src/multipart.ts:172]: can we remove the trailing coma?

File: src/multipart.ts [hunk 4]
```diff
@@ -149,37 +169,37 @@ function process(part: Part): Input {
...
     })
   } else {
     Object.defineProperty(input, 'name', {
       value: header[1].split('=')[1].replace(/"/g, ''),
       writable: true,
       enumerable: true,
-      configurable: true
+      configurable: true,
```
Comment (nachomazzara) [inline src/multipart.ts:194]: can we remove the trailing coma?

File: src/multipart.ts [hunk 5]
```diff
@@ -149,37 +169,37 @@ function process(part: Part): Input {
...
     })
   }
 
   Object.defineProperty(input, 'data', {
     value: Buffer.from(part.part),
     writable: true,
     enumerable: true,
-    configurable: true
+    configurable: true,
```
Comment (nachomazzara) [inline src/multipart.ts:202]: can we remove the trailing coma?

**Review by nachomazzara — CHANGES_REQUESTED** [review]
Hey! Thanks for the contribution. 

I've requested some formatting changes.

**Comment by simhnna** [comment]
> Hey! Thanks for the contribution.
> 
> I've requested some formatting changes.

I ran prettier after my changes since I saw config for it in the package.json 
Maybe the config needs to be updated? I can manually remove the commas

**Comment by crtag** [comment]
Such a good PR held by a comma which is a good practice to have in where it is now? The package is practically useless without this change.

**Comment by nachomazzara** [comment]
@simhnna the package lock maybe out of date. Can you send a resolve the conflicts so I can merge the PR?

**Comment by simhnna** [comment]
Sorry. had other work to do and then was on parental leave. Conflict resolved, commas removed and I updated the prettier config so it doesn't add the commas back

**Pull request merged**